### PR TITLE
Add control of single topic gc in inactive state.

### PR DESCRIPTION
**Pull request by zhanghaou**
**Motivation**
The brokerDeleteInactiveTopicsEnabled parameter is effect to all Topics, but in some cases like topic created by protocol handler, we need keep some topics not be deleted even if they're in inactive state.

**Modifications**
Add deleteWhileInactive parameter in AbstractTopic, and when create a new topic,  we can call topic.setDeleteWhileInactive(false) to keep the topic not be deleted by GC.

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractTopic.java [hunk 1]
```diff
@@ -423,5 +428,13 @@ public long getBytesOutCounter() {
         return getStats(false).bytesOutCounter;
     }
 
+    public boolean isBrokerDeleteInactiveTopicsEnabled() {
+        return brokerDeleteInactiveTopicsEnabled;
+    }
+
+    public void setBrokerDeleteInactiveTopicsEnabled(boolean brokerDeleteInactiveTopicsEnabled) {
```
Comment (merlimat) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractTopic.java:435]: How is this supposed to be activated? 

Instead, we need a namespace policy, attached to https://pulsar.apache.org/admin-rest-api/?version=2.5.2#operation/setAutoTopicCreation that adds the auto-deletion override configuration, on top of the auto-creation

Comment (codelipenghui) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractTopic.java:435]: @merlimat This is used by the protocol handlers who want to ensure some topics under the namespace does not delete by the broker. But do not want to change the namespace level policy. Now, we don't have a topic level policy, without this change, we must create a virtual producer or consume.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractTopic.java [hunk 2]
```diff
@@ -63,6 +63,9 @@
 
     protected volatile boolean isFenced;
 
+    // When set to false, this inactive topic can not be deleted
+    protected boolean brokerDeleteInactiveTopicsEnabled;
```
Comment (codelipenghui) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractTopic.java:67]: I think it's better to named `deleteWhileInactive`.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/nonpersistent/NonPersistentTopic.java [hunk 3]
```diff
@@ -830,6 +830,10 @@ public boolean isActive() {
 
     @Override
     public void checkGC(int maxInactiveDurationInSec, InactiveTopicDeleteMode deleteMode) {
+        if (brokerDeleteInactiveTopicsEnabled) {
```
Comment (codelipenghui) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/nonpersistent/NonPersistentTopic.java:833]:
```suggestion
        if (! brokerDeleteInactiveTopicsEnabled) {
```

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java [hunk 4]
```diff
@@ -1636,6 +1629,10 @@ private boolean hasBacklogs() {
 
     @Override
     public void checkGC(int maxInactiveDurationInSec, InactiveTopicDeleteMode deleteMode) {
+        if (brokerDeleteInactiveTopicsEnabled) {
```
Comment (codelipenghui) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java:1632]:
```suggestion
        if (!brokerDeleteInactiveTopicsEnabled) {
```

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractTopic.java (hunk 2 above)
Comment (zhanghaou) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractTopic.java]: Done.

**Comment by codelipenghui** [comment]
ping @merlimat please help review this PR again.

**Pull request merged**